sdk/ai/ai-inference-rest/src/tracingPolicy.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  type TracingClient,
+  type TracingContext,
+  type TracingSpan,
+  createTracingClient,
+} from "@azure/core-tracing";
+import { logger } from "./logger.js";
+import { getErrorMessage } from "@azure/core-util";
+import { tryProcessResponse, onStartTracing, tryProcessError, getSpanName } from "./traceHelper.js";
+import { PipelinePolicy, PipelineRequest, PipelineResponse, SendRequest } from "@azure/core-rest-pipeline";
+import { SDK_VERSION } from "./constants.js";
+
+/**
+ * The programmatic identifier of the tracingPolicy.
+ */
+export const tracingPolicyName = "inferenceTracingPolicy";
+
+/**
+ * A simple policy to create OpenTelemetry Spans for each request made by the pipeline
+ * that has SpanOptions with a parent.
+ * Requests made without a parent Span will not be recorded.
+ */
+export function tracingPolicy(): PipelinePolicy {
+
+  const tracingClient = tryCreateTracingClient();
```

Comment:
nit: You could just use `createTracingClient` - all packages use it fine its just that in core-rest-pipeline we wanted to be _extra_ defensive.